src/main/java/com/yelp/nrtsearch/server/luceneserver/BackupIndexRequestHandler.java

```diff
@@ -15,30 +15,52 @@
  */
 package com.yelp.nrtsearch.server.luceneserver;
 
+import com.google.gson.Gson;
 import com.yelp.nrtsearch.server.grpc.BackupIndexRequest;
 import com.yelp.nrtsearch.server.grpc.BackupIndexResponse;
 import com.yelp.nrtsearch.server.grpc.CreateSnapshotRequest;
-import com.yelp.nrtsearch.server.grpc.CreateSnapshotResponse;
 import com.yelp.nrtsearch.server.grpc.ReleaseSnapshotRequest;
 import com.yelp.nrtsearch.server.grpc.ReleaseSnapshotResponse;
+import com.yelp.nrtsearch.server.grpc.SnapshotId;
 import com.yelp.nrtsearch.server.utils.Archiver;
 import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import org.apache.lucene.util.IOUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class BackupIndexRequestHandler implements Handler<BackupIndexRequest, BackupIndexResponse> {
+  private static final String BACKUP_INDICATOR_FILE_NAME = "backup.txt";
   Logger logger = LoggerFactory.getLogger(BackupIndexRequestHandler.class);
   private final Archiver archiver;
+  private final Path archiveDirectory;
+  private final Path backupIndicatorFilePath;
+  private final Lock lock = new ReentrantLock();
```

Comment:
Good point. Making the lock static was resulting in some test failures. I have changed it so that the lock (made static) is only locked before creating the backup indicator file.